conans/client/generators/msbuild.py

```diff
@@ -0,0 +1,208 @@
+        condition = " And ".join("'$(%s)' == '%s'" % (k, v) for k, v in props if v)
+        return name.lower(), condition
+
+    def _multi(self, name_multi, name_conf, condition):
+        # read the existing mult_filename or use the template if it doesn't exist
+        multi_path = os.path.join(self.output_path, name_multi)
+        if os.path.isfile(multi_path):
+            content_multi = load(multi_path)
+        else:
+            content_multi = self.dep_props
+
+        # parse the multi_file and add a new import statement if needed
+        dom = minidom.parseString(content_multi)
+        import_group = dom.getElementsByTagName('ImportGroup')[0]
+        children = import_group.getElementsByTagName("Import")
+        for node in children:
+            if (name_conf == node.getAttribute("Project") and
+                    condition == node.getAttribute("Condition")):
+                # the import statement already exists
+                break
+        else:
+            # create a new import statement
+            import_node = dom.createElement('Import')
+            import_node.setAttribute('Condition', condition)
+            import_node.setAttribute('Project', name_conf)
+            # add it to the import group
+            import_group.appendChild(import_node)
+        content_multi = dom.toprettyxml()
+        content_multi = "\n".join(line for line in content_multi.splitlines() if line.strip())
+        return content_multi
+
+    def _general(self, name_general, deps, conf_name, condition):
+        # read the existing mult_filename or use the template if it doesn't exist
+        multi_path = os.path.join(self.output_path, name_general)
+        if os.path.isfile(multi_path):
+            content_multi = load(multi_path)
+        else:
+            content_multi = self.dep_props
+
+        # parse the multi_file and add a new import statement if needed
+        dom = minidom.parseString(content_multi)
+        import_group = dom.getElementsByTagName('ImportGroup')[0]
+        children = import_group.getElementsByTagName("Import")
+        for dep in deps:
+            conf_props_name = "conan_%s%s.props" % (dep, conf_name)
+            for node in children:
+                if (conf_props_name == node.getAttribute("Project") and
+                        condition == node.getAttribute("Condition")):
+                    # the import statement already exists
+                    break
+            else:
+                # create a new import statement
+                import_node = dom.createElement('Import')
+                import_node.setAttribute('Condition', condition)
+                import_node.setAttribute('Project', conf_props_name)
+                # add it to the import group
+                import_group.appendChild(import_node)
+        content_multi = dom.toprettyxml()
+        content_multi = "\n".join(line for line in content_multi.splitlines() if line.strip())
```

Comment:
Is it needed at all?

Reply:
Yes, otherwise the output has tons of blank lines

Reply:
There is `newl` argument to `toprettyxml`, it works, but we would need to clean the extra spaces/newlines before and after the XML tags.